modules/tinymce/src/themes/silver/main/ts/modes/Inline.ts

```diff
@@ -61,7 +61,16 @@ const setupEvents = (editor: Editor, targetElm: SugarElement, ui: InlineHeader,
     requestAnimationFrame(() => resizeContent(e));
   });
 
-  editor.on('ScrollWindow', () => ui.updateMode());
+  let lastScrollX = 0;
+  editor.on('ScrollWindow', () => {
+    const newScrollX = Scroll.get().left;
+    if (newScrollX !== lastScrollX) {
+      lastScrollX = newScrollX;
+      ui.update();
```

Comment:
Does `ScrollWindow` fire for every pixel? We might need to debounce this so it's not using a lot of CPU. `ui.update()` is quite a thorough and complex operation.

Reply:
How would you prefer to do it? I don't know if we have any existing functionality for it.

Reply:
A debounce? That's `Throttler.last()`. Set it to 33ms and see how it looks, if it's too janky maybe lower it a bit but not below 16.